### PR TITLE
GumGum Bid Adapter : add ORTB2 device data to request payload

### PR DESCRIPTION
**Pull request by jwrosewell**
## Type of change
- [x] Feature
- [x] Updated bidder adapter

## Description of change
This PR enhances the GumGum bid request by incorporating device data from the global ORTB2 object.

The device object has previously been populated by simplistic parsers, if at all, and was inaccurate as a result. Prebid now benefits from RTD modules such as 51Degrees that enrich all the device object fields including Apple iPhone model category and device ID. The PR enables GumGum’s users to benefit from device object improvements.

ORTB2 values have been mapped on to gumgum bid request parameter schema as follows:

- **ua**: ORTB2.device.ua
- **dnt**: ORTB2.device.dnt
- **os**: ORTB2.device.os
- **osv**: ORTB2.device.osv
- **dt**: ORTB2.device.devicetype
- **lang**: ORTB2.device.language
- **make**: ORTB2.device.make
- **model**: ORTB2.device.model
- **ppi**: ORTB2.device.ppi
- **pxratio**: ORTB2.device.pxratio
- **foddid**: (FiftyOne)51Degrees device ID

## Other information
cc: @MartinGumGum

### CODE REVIEW

**Comment by jwrosewell** [comment]
@MartinGumGum - this PR will ensure GumGum benefits from publishers efficiently populating the device object with enriched data including iPhone model categories and other features not obtainable via SUA and UA. As publishers progress rollout this becomes more critical.

File: modules/gumgumBidAdapter.js [hunk 1]
```diff
@@ -247,6 +247,39 @@ function _getFloor(mediaTypes, staticBidFloor, bid) {
...
+/**
+ * Retrieves the device data from the ORTB2 object
+ * @param {Object} ortb2Data ORTB2 object
+ * @returns {Object} Device data
+ */
+function _getDeviceData(ortb2Data) {
+  const _device = deepAccess(ortb2Data, 'device') || {};
+
+  // set device data params from ortb2
+  const _deviceRequestParams = {
```
Comment (patmmccann) [inline modules/gumgumBidAdapter.js:259]: @BohdanVV can you add ip and ipv6 to this?

Comment (BohdanVV) [reply modules/gumgumBidAdapter.js:259]: > @BohdanVV can you add ip and ipv6 to this?

Done 👌🏻

**Review by MartinGumGum — APPROVED** [review]
LGTM

**Comment by MartinGumGum** [comment]
@BohdanVV Circleci build has failed could you please rerun and merge?

**Comment by BohdanVV** [comment]
Hi @MartinGumGum, I don't have the necessary permissions to do that, but I'll make sure to ask someone who does. :)

**Comment by MartinGumGum** [comment]
Hi @patmmccann , maybe you can help us rerun circlei?  We don't have permission.

**Review by MartinGumGum — APPROVED** [review]

**Review by ChrisHuie — APPROVED** [review]

**Pull request merged**